ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleManager.java

```diff
@@ -151,73 +122,56 @@ boolean isAdPlayable(@NonNull String placement, @NonNull AdSize adSize) {
-        if (bannerAd != null) {
-            activeBannerAds.put(placement, bannerAd);
-        }
-
-        return bannerAd;
+        return Vungle.isInitialized() && Vungle.getValidPlacements().contains(placementId);
     }
 
-    VungleNativeAd getVungleNativeAd(String placement, AdConfig adConfig,
-                                     VungleListener vungleListener) {
-        Log.d(TAG, "getVungleNativeAd");
-
-        //Fetch new ad
-        VungleNativeAd bannerAd = Vungle.getNativeAd(placement, adConfig,
-                playAdCallback(vungleListener));
-        if (bannerAd != null) {
-            activeNativeAds.put(placement, bannerAd);
-        }
-
-        return bannerAd;
-    }
-
-    void cleanUpBanner(@NonNull String placementId, VungleNativeAd bannerAd) {
-        VungleNativeAd activeBannerAd = activeNativeAds.get(placementId);
-        if (activeBannerAd == bannerAd) {
-            //Remove ad
-            //We should do Report ad
-            Log.d(TAG, "cleanUpBanner # finishDisplayingAd");
-            bannerAd.finishDisplayingAd();
-            activeNativeAds.remove(placementId);
+    @Nullable
+    synchronized VungleBannerAdapter getBannerRequest(@NonNull String placementId, @Nullable String requestUniqueId) {
+        VungleBannerAdapter bannerRequest = mVungleBanners.get(placementId);
+        if (bannerRequest != null) {
+            String activeUniqueRequestId = bannerRequest.getUniquePubRequestId();
+            if (activeUniqueRequestId != null && requestUniqueId != null) {
+                if (activeUniqueRequestId.equals(requestUniqueId)) {
+                    Log.d(TAG, "Seems like Refresh: activeUniqueId: " + activeUniqueRequestId + " ###  RequestId: " + requestUniqueId);
+                } else {
+                    Log.d(TAG, "Does NOT seems like Refresh: activeUniqueId: " + activeUniqueRequestId + " ###  RequestId: " + requestUniqueId);
+                    return null;
+                }
+            } else if (activeUniqueRequestId != null) {
+                //Same placement: One banner is displaying with mediationExtras, but this banner request try to load without mediationExtras.
+                Log.d(TAG, "Does NOT seems like Refresh: activeUniqueId: " + activeUniqueRequestId + " ###  RequestId: null");
+                return null;
+            } else if (requestUniqueId != null) {
+                //Same placement: One banner is displaying without mediationExtras, but this banner request try to load with mediationExtras.
+                Log.d(TAG, "Does NOT seems like Refresh: activeUniqueId: null ###  RequestId: " + requestUniqueId);
+                return null;
+            } else {
+                if (bannerRequest.isLoading()) {
+                    Log.d(TAG, "Does NOT seems like Refresh: activeUniqueId: null ###  RequestId: null");
+                    return null;
+                } else {
+                    Log.d(TAG, "Seems like Refresh: activeUniqueId: null ###  RequestId: null");
```

Comment:
Is return missing here?  Non null bannerRequest will be returned.

Reply:
One case is none UUID banner ad auto-refresh.